docs/concepts/Scheduler/README.md

```diff
@@ -0,0 +1,89 @@
+# Fleet Scheduler
+
+The scheduler component is a vital element in Fleet workload scheduling. Its primary responsibility is to determine the
+schedule decision for a bundle of resources based on the latest `ClusterSchedulingPolicySnapshot`generated by the `ClusterResourcePlacement`.
+By default, the scheduler operates in batch mode, which enhances performance. In this mode, it binds a `ClusterResourceBinding`
+from a `ClusterResourcePlacement` to multiple clusters whenever possible.
+
+## Batch in nature
+
+Scheduling resources within a `ClusterResourcePlacement` involves more dependencies compared with scheduling pods within
+a deployment in Kubernetes. There are two notable distinctions:
+
+1. In a `ClusterResourcePlacement`, multiple replicas of resources cannot be scheduled on the same cluster, whereas pods
+belonging to the same deployment in Kubernetes can run on the same node.
+2. The `ClusterResourcePlacement` supports different placement types within a single object.
+
+These requirements necessitate treating the scheduling policy as a while and feeding it to the scheduler, as opposed to 
+handling individual pods like Kubernetes today. Specially:
+1. Scheduling the entire `ClusterResourcePlacement` at once enables us to increase the parallelism of the scheduler if
+needed.
+2. Supporting the `PickAll` mode would require generating the replica for each cluster in the fleet to scheduler. This

```

Comment:
If it is inefficient, what is the alternative approach for this? Is this a lead up to the statement about supporting PickN?